ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/trielog/TrieLogPruner.java

```diff
@@ -99,7 +99,7 @@ private int preloadQueue() {
     }
   }
 
-  public void addToPruneQueue(final long blockNumber, final Hash blockHash) {
+  public synchronized void addToPruneQueue(final long blockNumber, final Hash blockHash) {
```

Comment:
could you explain why this modification is needed ? I think if it was not needed on mainnet, I think it should be the same on qbft ?

Reply:
I think it is an existing bug that hasn't been hit on mainnet because the rate of receiving and handling new trie-logs from peers isn't fast enough to hit the threading issue.

I was getting `ConcurrentModificationExceptions` on the `trieLogBlocksAndForksByDescendingBlockNumber` map when a QBFT node was syncing from several peers due to `addToPruneQueue` modifying the map while `pruneFromQueue` was iterating over it.

Since it hasn't been hit on mainnet before, I'm assuming the making the methods `synchronized` isn't going to affect mainnet syncing performance at all.

Reply:
If it's useful I can try recreating it to show the full `ConcurrentModificationException` stack, but it was definitely an intermittent (but reliable) exception once we had got the basic syncing logic working.

Reply:
Sounds good. I just ping @siladu who has coded this part and can verify that we are good